src/Microsoft.PowerShell.Commands.Utility/commands/utility/group-object.cs

```diff
@@ -238,25 +269,26 @@ public SwitchParameter NoElement
         /// Utility function called by Group-Object to create Groups.
         /// </summary>
         /// <param name="currentObjectEntry">Input object that needs to be grouped.</param>
-        /// <param name="noElement">true if we are not accumulating objects</param>
+        /// <param name="noElement">True if we are not accumulating objects.</param>
         /// <param name="groups">List containing Groups.</param>
         /// <param name="groupInfoDictionary">Dictionary used to keep track of the groups with hash of the property values being the key.</param>
         /// <param name="orderByPropertyComparer">The Comparer to be used while comparing to check if new group has to be created.</param>
-        internal static void DoGrouping(OrderByPropertyEntry currentObjectEntry, bool noElement, List<GroupInfo> groups, Dictionary<object, GroupInfo> groupInfoDictionary,
+        internal static void DoGrouping(
+            OrderByPropertyEntry currentObjectEntry,
+            bool noElement,
+            List<GroupInfo> groups,
+            Dictionary<object, GroupInfo> groupInfoDictionary,
             OrderByPropertyComparer orderByPropertyComparer)
         {
-            if (currentObjectEntry != null && currentObjectEntry.orderValues != null && currentObjectEntry.orderValues.Count > 0)
+            var currentObjectorderValues = currentObjectEntry.orderValues;
+            if (currentObjectorderValues != null && currentObjectorderValues.Count > 0)
             {
-                object currentTupleObject = PSTuple.ArrayToTuple(currentObjectEntry.orderValues.ToArray());
+                object currentTupleObject = PSTuple.ArrayToTuple(currentObjectorderValues.ToArray());
 
-                GroupInfo currentGroupInfo = null;
-                if (groupInfoDictionary.TryGetValue(currentTupleObject, out currentGroupInfo))
+                if (groupInfoDictionary.TryGetValue(currentTupleObject, out var currentGroupInfo))
```

Comment:
What's the use case for currentGroupInfo being null in the groupInfoDictionary?  If it is expected, a comment is needed; otherwise, it looks like currentObjectEntry.inputObject is dropped on the floor.

Reply:
I don't think that is a case that will ever occur. Just defensive programming.

Both cases are behaving as the original implementation. 

On could argue for removing the null check as it is a condition that shouldn't occur, and if it does, it is a bug that should be fixed.

Reply:
>  if it does, it is a bug that should be fixed.

So should we have runtime check and throw?